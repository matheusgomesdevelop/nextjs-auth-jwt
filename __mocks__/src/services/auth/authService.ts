import { AuthCredentialSchema, AuthResponseSchema } from '@/schemas/AuthSchema';

export const mockLoginEmptyResponse: AuthResponseSchema = {
    jwt: '',
    user: {
        id: 0,
        username: '',
        email: '',
        provider: '',
        confirmed: false,
        blocked: false,
        createdAt: '',
        updatedAt: '',
    },
};
