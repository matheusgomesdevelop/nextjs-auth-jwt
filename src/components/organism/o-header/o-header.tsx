import styles from "./o-header.module.scss";

// Atoms
import { A_Avatar } from "@/components/atoms";

export const O_Header = () => {
  return (
    <header data-testid="o-header" className={styles.o_header}>
      <A_Avatar image={"/images/a-avatar.jpeg"} label={""} shape="circle" />
    </header>
  );
};
