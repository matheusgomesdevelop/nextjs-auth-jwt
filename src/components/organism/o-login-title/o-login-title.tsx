// Atoms
import { A_Avatar, A_Title, A_Text } from "@/components/atoms";

// StyleSheet
import styles from "./o-login-title.module.css";

export interface I_OLoginTitle {
  image: string;
  title: string;
  username: string;
}

export const O_LoginTitle = ({ image, title, username }: I_OLoginTitle) => {
  return (
    <div data-testid="o-login-title" className={styles.o_loginTitle}>
      <A_Avatar
        image={image}
        label={username[0]}
        size="normal"
        shape="circle"
      />
      <A_Title variant="h1" className={styles.a_title}>
        {title}
      </A_Title>
      <A_Text variant="fwReg-fs16-gray500">{username}</A_Text>
    </div>
  );
};
