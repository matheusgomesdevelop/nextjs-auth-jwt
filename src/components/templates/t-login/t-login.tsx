
// Organism and I_OTypes
import {
  I_OLoginForm,
  O_LoginForm,
} from "@/components/organism/o-login-form/o-login-form";
import {
  I_OLoginTitle,
  O_LoginTitle,
} from "@/components/organism/o-login-title/o-login-title";

// StyleSheet
import styles from "./t-login.module.css";

interface I_TLogin {
  loginTitleProps: I_OLoginTitle;
  loginFormProps: I_OLoginForm;
}

export const T_Login = ({ loginTitleProps, loginFormProps }: I_TLogin) => {
  const { image, title, username } = loginTitleProps;
  const { onSubmit, errors } = loginFormProps;

  return (
    <div data-testid="t-login" className={styles.t_login}>
      <O_LoginTitle image={image} title={title} username={username} />
      <O_LoginForm onSubmit={onSubmit} errors={errors} />
    </div>
  );
};
