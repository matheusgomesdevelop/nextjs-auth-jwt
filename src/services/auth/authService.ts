import axios, { AxiosResponse } from "axios";

// Config
import { STRAPI_API_URL } from "@/config";

interface I_LoginCredentials {
  identifier: string;
  password: string;
}

interface I_UserData {
  id: number;
  username: string;
  email: string;
  provider: string;
  confirmed: boolean;
  blocked: number;
  createdAt: string;
  updatedAt: string;
}

interface I_LoginResponse {
  jwt: string;
  user: I_UserData;
}

interface I_AuthService {
  login: (body: I_LoginCredentials) => Promise<AxiosResponse<I_LoginResponse>>;
}

export const authService: I_AuthService = {
  login: async (body: { identifier: string; password: string }) => {
    const url = `${STRAPI_API_URL}/auth/local`;

    try {
      const response: AxiosResponse<I_LoginResponse> = await axios.post(url, body);

      return response;
    } catch (error) {
      throw error;
    }
  },
};
