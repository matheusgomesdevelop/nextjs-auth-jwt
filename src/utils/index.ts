import { messages } from "./config";

export { messages };
