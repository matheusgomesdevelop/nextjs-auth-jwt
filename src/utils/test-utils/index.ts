import { AppRouterContextProviderMock } from "./providers/app-router-ctx-provider-mock";

export { AppRouterContextProviderMock };
