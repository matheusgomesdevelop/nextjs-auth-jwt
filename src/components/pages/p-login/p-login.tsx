// StyleSheet
import styles from "./p-login.module.css";

// Interfaces
import { I_TLogin } from "@/components/templates/t-login/t-login";

// Templates
import { T_Login } from "@/components/templates";

interface I_PHome {
  templateLoginProps: I_TLogin;
}

export const P_Home = ({ templateLoginProps }: I_PHome) => {
  const { loginTitleProps, loginFormProps } = templateLoginProps;

  return (
    <main data-testid="p-home" className={styles.p_home}>
      <T_Login
        loginTitleProps={loginTitleProps}
        loginFormProps={loginFormProps}
      />
    </main>
  );
};
