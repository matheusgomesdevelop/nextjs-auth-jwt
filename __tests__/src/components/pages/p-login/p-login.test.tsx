import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

// Interfaces
import { I_OLoginTitle } from '@/components/organism/o-login-title/o-login-title';

// Pages
import { P_Home } from '../../../../../src/components/pages/p-login/p-login';

// Utils
import { AppRouterContextProviderMock } from '@/core/utils/test-utils';
import { I_OHeader } from '@/components/organism/header/o-header';
import { mockLoginResponse } from '../../../../../__mocks__/src/services/auth/authService';
import { I_TLogin } from '@/components/templates/Login/t-login';

describe('Deve renderizar a pagina de login, corretamente', () => {
    const handleLoginForm = jest.fn();
    const push = jest.fn();

    // Organism: Header
    const o_headerProps: I_OHeader = {
        link: 'https://github.com/matheusgrodrigues',
        image: '/images/a-avatar.jpeg',
    };

    // Organism: LoginTitle
    const o_loginTitle: I_OLoginTitle = {
        image: '/images/a-avatar.jpeg',
        title: 'Acesse sua conta',
        userSession: { data: { session: mockLoginResponse.user }, error: false, loading: false },
    };

    // Template: LoginForm
    const t_loginProps: I_TLogin = {
        o_loginTitle,
        handleLoginForm,
    };

    // Organism: Footer
    const o_footerProps = {
        name: 'matheusgomesdev',
        site: 'https://matheusgomesdev.com.br',
        github: 'https://github.com/matheusgrodrigues',
        linkedin: 'https://www.linkedin.com/in/matheusgomes/',
    };

    beforeEach(() => {
        render(
            <AppRouterContextProviderMock router={{ push }}>
                <P_Home
                    o_headerProps={o_headerProps}
                    t_loginProps={{
                        o_loginTitle,
                        handleLoginForm,
                    }}
                    o_footerProps={o_footerProps}
                />
            </AppRouterContextProviderMock>
        );
    });
    // ============================================
    it('Deve renderizar o t-login', () => {
        // Arrange -: beforeEach

        // Act
        const get_t_login = screen.getByTestId('t-login');

        // Assert
        expect(get_t_login).toBeInTheDocument();
    });
    // ============================================

    it('Deve preservar a estrutura visual da p-login', () => {
        const get_p_login = renderer
            .create(
                <AppRouterContextProviderMock router={{ push }}>
                    <P_Home
                        o_headerProps={o_headerProps}
                        t_loginProps={{
                            o_loginTitle,
                            handleLoginForm,
                        }}
                        o_footerProps={o_footerProps}
                    />
                </AppRouterContextProviderMock>
            )
            .toJSON();

        expect(get_p_login).toMatchSnapshot();
    });
    // ============================================
});
