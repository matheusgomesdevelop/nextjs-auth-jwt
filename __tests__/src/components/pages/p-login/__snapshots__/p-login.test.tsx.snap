// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar a pagina de login, corretamente Deve preservar a estrutura visual da p-login 1`] = `
<main
  className="p_home"
  data-testid="p-home"
>
  <header
    className="o_header"
    data-testid="o-header"
  >
    <div
      className="p-avatar p-component p-avatar-image p-avatar-circle"
      data-pc-name="avatar"
      data-pc-section="root"
      data-testid="a-avatar"
      style={{}}
    >
      <img
        alt="avatar"
        data-pc-section="image"
        onError={[Function]}
        src="/images/a-avatar.jpeg"
      />
    </div>
  </header>
  <div
    className="t_login"
    data-testid="t-login"
    id="t-login"
  >
    <div
      className="o_loginTitle"
      data-testid="o-login-title"
    >
      <i
        className="a_icon pi pi-lock"
      />
      <h2
        className="a_title__h2"
        data-testid="a-title"
      />
      <p
        className="a_text__fwReg_fs16_gray500"
        data-testid="a-text"
      >
        Estamos felizes em vê-lo novamente! Insira suas credenciais para entrar.
      </p>
    </div>
    <form
      className="o_form_login"
      data-testid="o-login-form"
      onSubmit={[Function]}
    >
      <div
        className="m_input_label"
        data-testid="m-input-with-label-email"
      >
        <label
          className="a_label"
          data-testid="a-label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="p-inputtext p-component"
          data-pc-name="inputtext"
          data-pc-section="root"
          data-testid="a-input-email"
          name="email"
          onBeforeInput={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          onKeyDown={[Function]}
          onPaste={[Function]}
          placeholder="Endereço de e-mail"
          type="email"
          value=""
        />
      </div>
      <div
        className="m_input_label"
        data-testid="m-input-with-label-password"
      >
        <label
          className="a_label"
          data-testid="a-label"
          htmlFor="password"
        >
          Senha
        </label>
        <input
          className="p-inputtext p-component"
          data-pc-name="inputtext"
          data-pc-section="root"
          data-testid="a-input-password"
          name="password"
          onBeforeInput={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          onKeyDown={[Function]}
          onPaste={[Function]}
          placeholder="Digite sua senha"
          type="password"
          value=""
        />
      </div>
      <div
        className="m_checkboxWithLabel"
        data-testid="m-checkboxWithLabel"
      >
        <div
          className="p-checkbox p-component"
          data-pc-name="checkbox"
          data-pc-section="root"
          data-testid="a-checkbox"
          id="a-checkbox"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={null}
          onMouseDown={null}
          style={{}}
        >
          <div
            className="p-hidden-accessible"
            data-pc-section="hiddeninputwrapper"
          >
            <input
              data-pc-section="hiddeninput"
              defaultChecked={false}
              disabled={false}
              id={null}
              name="manter_logado"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={false}
              required={false}
              tabIndex={null}
              type="checkbox"
            />
          </div>
          <div
            className="p-checkbox-box"
            data-p-disabled={false}
            data-p-focus={false}
            data-p-highlight={false}
            data-pc-section="input"
          />
        </div>
        <label
          className="a_label"
          data-testid="a-label"
          onClick={[Function]}
        >
          Mantenha-me conectado.
        </label>
      </div>
      <button
        className="a_button__gradient p-button p-component"
        data-pc-name="button"
        data-pc-section="root"
        data-testid="a-button"
        disabled={false}
        type="submit"
      >
        Entrar
      </button>
    </form>
  </div>
</main>
`;
